src/domain/user-group/user-group.service.ts

```diff
@@ -73,10 +74,71 @@ export class UserGroupService {
         LogContext.COMMUNITY
       );
 
+    // Cannot remove restricted groups
+    if (await this.isRestricted(group))
+      throw new ForbiddenException(
+        `Unable to remove User Group with the specified ID: ${group.id}; restricted group: ${group.name}`,
+        LogContext.COMMUNITY
+      );
+
     await this.groupRepository.remove(group);
     return true;
   }
 
+  // Note: explicitly do not support updating of email addresses
+  async updateUserGroup(
+    groupID: number,
+    userGroupInput: UserGroupInput
+  ): Promise<IUserGroup> {
+    const group = await this.groupRepository.findOne({
+      where: { id: groupID },
+    });
+    if (!group)
+      throw new EntityNotFoundException(
+        `Unable to update group with ID: ${groupID}`,
+        LogContext.COMMUNITY
+      );
+    // Cannot rename restricted groups
+    const newName = userGroupInput.name;
+    if (newName && newName.length > 0 && newName !== group.name) {
+      // group being renamed; check if allowed
+      if (await this.isRestricted(group)) {
+        throw new ForbiddenException(
```

Comment:
Forbidden = Unauthorized. This is unrelated to auth issues. Use ValidationException instead.